src/common/Utilities/Util.h

```diff
@@ -772,171 +762,181 @@ class EventMap
+* @param delay Amount of delay.
+*/
+void DelayEvents(uint32 delay)
+{
+    _time = delay < _time ? _time - delay : 0;
+}
+
+void DelayEventsToMax(uint32 delay, uint32 group)
+{
+    for (EventStore::iterator itr = _eventMap.begin(); itr != _eventMap.end();)
     {
-        for (EventStore::iterator itr = _eventMap.begin(); itr != _eventMap.end();)
+        if (itr->first < _time + delay && (group == 0 || ((1 << (group + 15)) & itr->second)))
         {
-            if (itr->first < _time + delay && (group == 0 || ((1 << (group + 15)) & itr->second)))
+            if (itr->first < _time+delay && (group == 0 || ((1 << (group + 15)) & itr->second)))
             {
-                if (itr->first < _time+delay && (group == 0 || ((1 << (group + 15)) & itr->second)))
-                {
-                    ScheduleEvent(itr->second, delay);
-                    _eventMap.erase(itr);
-                    itr = _eventMap.begin();
-                }
-                else
-                {
-                    ++itr;
-                }
+                ScheduleEvent(itr->second, delay);
+                _eventMap.erase(itr);
+                itr = _eventMap.begin();
             }
             else
+            {
                 ++itr;
+            }
         }
+        else
+            ++itr;
     }
+}
 
-        /**
-        * @name DelayEvents
-        * @brief Delay all events of the same group.
-        * @param delay Amount of delay.
-        * @param group Group of the events.
-        */
-        void DelayEvents(uint32 delay, uint32 group)
+    /**
+    * @name DelayEvents
+    * @brief Delay all events of the same group.
+    * @param delay Amount of delay.
+    * @param group Group of the events.
+    */
+    void DelayEvents(uint32 delay, uint32 group)
+    {
+        if (group > 8 || Empty())
         {
-            if (group > 8 || Empty())
-            {
-                return;
-            }
+            return;
+        }
 
-        EventStore delayed;
+    EventStore delayed;
 
-        for (EventStore::iterator itr = _eventMap.begin(); itr != _eventMap.end();)
+    for (EventStore::iterator itr = _eventMap.begin(); itr != _eventMap.end();)
+    {
+        if (!group || (itr->second & (1 << (group + 15))))
         {
             if (!group || (itr->second & (1 << (group + 15))))
             {
-                if (!group || (itr->second & (1 << (group + 15))))
-                {
-                    delayed.insert(EventStore::value_type(itr->first + delay, itr->second));
-                    _eventMap.erase(itr++);
-                }
-                else
-                {
-                    ++itr;
-                }
+                delayed.insert(EventStore::value_type(itr->first + delay, itr->second));
+                _eventMap.erase(itr++);
             }
             else
+            {
                 ++itr;
+            }
         }
+        else
+            ++itr;
```

Comment:
Should be better merge all `if`s conditions into one and put only one else with `++itr;` ?